File: 2095_Delete_the_Middle_Node_of_a_Linked_List/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteMiddle(self, h: Optional[ListNode]) -> Optional[ListNode]:
        if not h or not h.next:
            return None
        s = h
        f = h.next.next
        while f and f.next:
            s = s.next
            f = f.next.next
        s.next = s.next.next
        return h

File: 2095_Delete_the_Middle_Node_of_a_Linked_List/solution.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* deleteMiddle(struct ListNode* h) {
    if (!h || !h->next) return NULL;
    struct ListNode *s = h, *f = h->next->next;
    while (f && f->next) {
        s = s->next;
        f = f->next->next;
    }
    s->next = s->next->next;
    return h;
}